pkg/cpuallocator/allocator.go

```diff
@@ -366,30 +420,43 @@ func allocateCpus(from *cpuset.CPUSet, cnt int) (cpuset.CPUSet, error) {
 		a := NewCPUAllocator(nil)
 		a.from = from.Clone()
 		a.cnt = cnt
+		a.preferred = preferred
 
 		result, err, *from = a.allocate(), nil, a.from.Clone()
 
-		a.Debug("allocateCpus(#%s, %d) => #%s", from.Union(result).String(), cnt, result)
+		a.Debug("%d cpus from #%v (preferring #%v) => #%v", cnt, from.Union(result), preferred, result)
 	}
 
 	return result, err
 }
 
 // AllocateCpus allocates a number of CPUs from the given set.
-func AllocateCpus(from *cpuset.CPUSet, cnt int) (cpuset.CPUSet, error) {
-	result, err := allocateCpus(from, cnt)
+func AllocateCpus(from *cpuset.CPUSet, cnt int, preferHighPrio bool) (cpuset.CPUSet, error) {
+	preferred := system.priorityCpus
+	if !preferHighPrio {
+		// Try to avoid high priority cpus
+		preferred = from.Difference(system.priorityCpus)
+	}
+
+	result, err := allocateCpus(from, cnt, preferred)
 	return result, err
 }
 
 // ReleaseCpus releases a number of CPUs from the given set.
-func ReleaseCpus(from *cpuset.CPUSet, cnt int) (cpuset.CPUSet, error) {
+func ReleaseCpus(from *cpuset.CPUSet, cnt int, preferHighPrio bool) (cpuset.CPUSet, error) {
 	var oset cpuset.CPUSet
 
 	if debug {
 		oset = from.Clone()
 	}
 
-	result, err := allocateCpus(from, from.Size()-cnt)
+	preferred := system.priorityCpus
+	if !preferHighPrio {
```

Comment:
nitpick: not that it would matter because none of the policies do partial releases of CPUs... but if the logic by the caller for deducing preferHighPrio for AllocateCpus is to take it from an optional annotation, then the caller needs to invert that logic here. IOW, if it allocates with preferHighPrio v, then it would need to pass !v to ReleaseCpus for shrinking the container.

Reply:
Yeah, that's true. Any suggestions how to fix the logic/naming? I seem to be quite blind to that at this time of the week and day 😄 I think it sort of does what it "promises" i.e. releases high prio cpus if `preferHighPrio` is true. But OTOH, I understand the inherent risk of confusion, too

Reply:
I'm really not sure if it can be confusing for anybody else than myself. I mentioned just in case somebody else might want take a look at it and see how they feel. I honestly suspect that this really might be a neurotic personality disorder vs. normal human beings thing. Let's not touch this unless somebody else gets confused.